Sources/KlaviyoCore/KlaviyoEnvironment.swift

```diff
@@ -163,41 +135,64 @@ public struct KlaviyoEnvironment {
         randomInt: { Int.random(in: 0...10) },
         raiseFatalError: { msg in
             #if DEBUG
-            fatalError(msg)
+                fatalError(msg)
             #endif
         },
         emitDeveloperWarning: { runtimeWarn($0) },
-        networkSession: createNetworkSession,
         apiURL: { KlaviyoEnvironment.productionHost },
         encodeJSON: { encodable in try encoder.encode(encodable) },
         decoder: DataDecoder.production,
         uuid: { UUID() },
         date: { Date() },
         timeZone: { TimeZone.autoupdatingCurrent.identifier },
-        appContextInfo: { AppContextInfo() },
         klaviyoAPI: KlaviyoAPI(),
         timer: { interval in
-            Timer.publish(every: interval, on: .main, in: .default)
-                .autoconnect()
-                .eraseToAnyPublisher()
-        },
-        SDKName: KlaviyoEnvironment.getSDKName,
-        SDKVersion: KlaviyoEnvironment.getSDKVersion)
+            AsyncStream { continuation in
+                let timerActor = TimerActor()
+                Task {
+                    // Start the timer via the TimerActor
+                    #if swift(>=6)
+                    timerActor.startTimer(interval: interval, continuation: continuation)
+                    #else
+                    await timerActor.startTimer(interval: interval, continuation: continuation)
+                    #endif
+                }
+
+                continuation.onTermination = { _ in
+                    // Stop the timer when the stream terminates
+                    Task {
+                        #if swift(>=6)
+                        timerActor.stopTimer()
+                        #else
+                        await timerActor.stopTimer()
+                        #endif
+                    }
+                }
+            }
+        }, appContextInfo: {
+            #if swift(>=6)
+            getDefaultAppContextInfo()
+            #else
+            await getDefaultAppContextInfo()
+            #endif
+        })
 }
 
-public var networkSession: NetworkSession!
+@MainActor var networkSession: NetworkSession? = nil
+
+@MainActor
 public func createNetworkSession() -> NetworkSession {
     if networkSession == nil {
         networkSession = NetworkSession.production
     }
-    return networkSession
+    return networkSession!
 }
 
 public enum KlaviyoDecodingError: Error {
     case invalidType
 }
 
-public struct DataDecoder {
+public struct DataDecoder: Sendable {
```

Comment:
In practice I'm not sure how much it matters, but I think it's not technically safe to mark this as `Sendable` because the `JSONDecoder` reference could in theory be mutated by a different thread

Reply:
Yah good point it won't be but maybe I can mark unchecked sendable so it's more honest.

Reply:
Yah some limited research indicates this should be safe but maybe we could experiment with just recreating an instance on every decode/encode. Sounds like they are cheap to make.

Reply:
Yup of course! I think effectively I would have a property on here that is a function which returns a jsondecoder.

Reply:
made it unchecked for now.